src/drt/src/io/io_parser_helper.cpp

```diff
@@ -45,6 +45,33 @@ using namespace boost::polygon::operators;
 using Rectangle = boost::polygon::rectangle_data<int>;
 namespace gtl = boost::polygon;
 
+bool io::Parser::checkPinsAboveTopRoutingLayer()
+{
+  if (design_->getTopBlock()) {
+    for (const auto& bTerm : design_->getTopBlock()->getTerms()) {
+      if (bTerm->getNet() && !bTerm->getNet()->isSpecial()) {
+        if (bTerm->isAboveTopLayer()) {
+          return true;
+        }
+      }
+    }
+    for (const auto& inst : design_->getTopBlock()->getInsts()) {
+      for (const auto& iTerm : inst->getInstTerms()) {
+        if (iTerm->getNet() && !iTerm->getNet()->isSpecial()) {
+          for (const auto& pin : iTerm->getTerm()->getPins()) {
+            for (const auto& fig : pin->getFigs()) {
+              if (((frShape*) (fig.get()))->getLayerNum() > TOP_ROUTING_LAYER) {
+                return true;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  return false;
+}
+
```

Comment:
I think there is a better approach for this. We should find the top_pin_layer and check below against the max(TOP_ROUTING_LAYER, top_pin_layer). We don't need to get default vias for layer met10 when the top routing layer is met5 and the top_pin_layer is met7.